identity-wallet/src/state/persistence.rs

```diff
@@ -56,6 +56,18 @@ pub fn clear_assets_tmp_folder() -> Result<(), AppError> {
     Ok(())
 }
 
+/// Clears the `/assets` folder inside the system-specific data directory.
+/// This is only used when resetting the app to factory defaults.
+pub fn clear_all_assets() -> Result<(), AppError> {
+    let assets_dir = ASSETS_DIR.lock().unwrap().as_path().to_owned();
```

Comment:
Since this is a lock, don't we need to make this async and use await? Because it can technically fail.